driver-core/src/main/com/mongodb/internal/operation/WriteOperation.java

```diff
@@ -24,6 +25,12 @@
  * <p>This class is not part of the public API and may be removed or changed at any time</p>
  */
 public interface WriteOperation<T> {
+
+    /**
+     * @return the timeout settings for this operation
+     */
+    TimeoutSettings getTimeoutSettings();
```

Comment:
I think we talked about removing this from Read/WriteOperation and instead passing it directly to the operation executor. Did we reject that idea or is it still on the table?

Reply:
I think the plan is to do it in https://jira.mongodb.org/browse/JAVA-5176